client/src/PlayerStack.lua

```diff
@@ -1591,22 +1591,30 @@ function PlayerStack:getAttackPatternData()
   return data, state
 end
 
-function PlayerStack.updateDangerBounce(self)
+-- calculate which columns should bounce
+function PlayerStack:updateDangerBounce()
   if self.engine.puzzle then
     return
   end
 
-  -- calculate which columns should bounce
+  -- reset state
   self.danger = false
-  local panelRow = self.engine.panels[self.engine.height - 1]
-  for idx = 1, self.engine.width do
-    if panelRow[idx]:dangerous() then
-      self.danger = true
-      self.danger_col[idx] = true
-    else
-      self.danger_col[idx] = false
+  for column = 1, self.engine.width do
+    self.danger_col[column] = false
+  end
+
+  for row = self.engine.height - 1, self.engine.height do
+    local panelRow = self.engine.panels[row]
+    if panelRow then
+      for idx = 1, self.engine.width do
+        if panelRow[idx]:dangerous() then
+          self.danger = true
+          self.danger_col[idx] = true
+        end
+      end
```

Comment:
There was a bug here that caused garbage in row 12 to not be considered for danger col even if the stack was smoothly raised for row 12 to be in play.
You can confirm the behaviour with/without this fix at the end of this replay where column 3 does not change to the panic frames even though the row 12 garbage covers column 3 to 6.
[v048-2025-03-17-12-07-25-Shogundaris-L1-vsSelf.json](https://github.com/user-attachments/files/19287965/v048-2025-03-17-12-07-25-Shogundaris-L1-vsSelf.json)